new-dti-website/components/team/TeamAbout.tsx

```diff
@@ -0,0 +1,264 @@
+
+  const polarToRect = (angle: number, radius: number) => {
+    const newX = -radius * Math.cos(angle + Math.PI / 2);
+    const newY = -radius * Math.sin(angle + Math.PI / 2);
+    return [newX, newY];
+  };
+
+  const pointAsString = (point: number[]) => point.join(' ');
+  const scale = (point: number[], scalar: number) => point.map((x) => scalar * x);
+
+  return (
+    <svg height={height} width={width} viewBox="-200 -200 400 400">
+      {Object.keys(roleStats).map((role) => {
+        const percentage = roleStats[role].people / allMembers.length;
+        const theta = 2 * Math.PI * percentage;
+
+        const arcPoint1 =
+          role === chartSection ? scale(previousPoint, hoverRadius / radius) : previousPoint;
+        const currentRadius = role === chartSection ? hoverRadius : radius;
+
+        totalAngle += theta;
+        const arcPoint2 = polarToRect(totalAngle, role === chartSection ? hoverRadius : radius);
+
+        const nextPoint = polarToRect(totalAngle, radius);
+        const textLocation = polarToRect((2 * totalAngle - theta) / 2, (3 * radius) / 4);
+        previousPoint = nextPoint;
+
+        return (
+          <g
+            key={role}
+            onMouseEnter={() => setChartSection(role)}
+            onMouseLeave={() => setChartSection(undefined)}
+          >
+            <path
+              d={`M 0 0 L ${pointAsString(arcPoint1)} A ${currentRadius} ${currentRadius} 0 ${
+                theta > Math.PI ? `1` : `0`
+              } 1 ${pointAsString(arcPoint2)} L 0 0`}
+              fill={roleStats[role].color}
+            />
+            <text
+              x={textLocation[0]}
+              y={textLocation[1]}
+              className={`font-bold text-lg ${
+                role === 'developer' || role === 'lead' ? 'fill-white' : 'fill-black'
+              }`}
+              style={{ textAnchor: 'middle' }}
+            >
+              {`${Math.round(percentage * 100)}%`}
+            </text>
+          </g>
+        );
+      })}
+    </svg>
+  );
+};
+
+const TeamStatistics = () => {
+  const [chartSection, setChartSection] = useState<string | undefined>(undefined);
+  const { width } = useScreenSize();
+
+  const allMembers = FA23Members.members as IdolMember[];
+
+  const roleStats: roleStatistics = {
+    designer: {
+      name: 'Design',
+      color: '#FFBCBC',
+      people: 0,
+      majors: new Set(),
+      colleges: new Set()
+    },
+    developer: {
+      name: 'Development',
+      color: '#D63D3D',
+      people: 0,
+      majors: new Set(),
+      colleges: new Set()
+    },
+    pm: { name: 'Product', color: '#FFFFFF', people: 0, majors: new Set(), colleges: new Set() },
+    business: {
+      name: 'Business',
+      color: '#B7B7B7',
+      people: 0,
+      majors: new Set(),
+      colleges: new Set()
+    },
+    lead: { name: 'Leads', color: '#484848', people: 0, majors: new Set(), colleges: new Set() }
+  };
+
+  allMembers.forEach((member: IdolMember) => {
```

Comment:
generally discouraged to write this imperatively like this for each render. Could you potentially write it functionally? (with an `.map(...)` + array, then convert to set)?